File: Java/Guias/guia3/ej9_Figures/Rectangle.java
package Guias.guia3.ej9_Figures;

public class Rectangle extends Figure {
    private Point topLeft;
    private Point bottomRight;

    public Rectangle(Point topLeft, Point bottomRight){
        this.topLeft = topLeft;
        this.bottomRight = bottomRight;
    }

    public double base(){
        return Math.abs(bottomRight.getX() - topLeft.getX());
    }
     public double height(){
        return Math.abs(topLeft.getY() - bottomRight.getY());
     }

     public double area(){
        return base() * height();
     }

     public double perimeter(){
        return 2*(base() + height());
     }


     @Override
     public String toString(){
        return "Rectangle [ %s , %s]".formatted(topLeft, bottomRight);
     }
}

File: RUBY/Guias/Guia8/Ej4_Figures/rectangle.rb
# frozen_string_literal: true
require_relative 'figure'
class Rectangle < Figure
  def initialize(p1, p2)
    @p1 = p1
    @p2 = p2
  end

  def base
    (@p2.x - @p1.x).abs
  end
  def height
    (@p2.y - @p1.y).abs
  end

  def perimeter
    2*height + 2*base
  end

  def area
    height*base
  end

  def to_s
    "Rectangulo [ #{@p1} , #{@p2} ]"
  end
end
